### PR TITLE
Reworked Vert.x Integration

### LINKED ISSUE #112: Modularization: http-vertx module
This module implements the [HTTP Protocol Binding](https://github.com/cloudevents/spec/blob/v1.0/http-protocol-binding.md) with vertx.

Actually, we have this module, but it needs refactorings.

- `http-vertx`

__This module depend on [api](https://github.com/cloudevents/sdk-java/issues/111) one.__

When someone wants the http binding to create or read events with vertx, they add this module and they concern about potential conflicts with pre-existing dependencies in their projects.

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Fix #112

Supersede #107, #104, #116

### CODE REVIEW

**Comment by matejvasek** [comment]
While testing I found out that headers are case sensitive. For instance if I `curl` with `-H "Ce-Id: 42"` the id is not read. Maybe we should pre-process headers with `toLowerCase()`.

**Comment by slinkydeveloper** [comment]
That's weird, `MultiMap#get` should be case-insensitive

**Comment by slinkydeveloper** [comment]
@matejvasek can you post the test code you wrote?

File: http/vertx/src/main/java/io/cloudevents/http/vertx/impl/BinaryVertxMessageImpl.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        return Encoding.BINARY;
+    }
+
+    @Override
+    public <T extends BinaryMessageVisitor<V>, V> V visit(BinaryMessageVisitorFactory<T, V> visitorFactory) throws MessageVisitException, IllegalStateException {
+        BinaryMessageVisitor<V> visitor = visitorFactory.createBinaryMessageVisitor(this.version);
+
+        // Grab from headers the attributes and extensions
+        this.headers.forEach(e -> {
+            if (e.getKey().startsWith(CE_PREFIX)) {
```
Comment (matejvasek) [inline http/vertx/src/main/java/io/cloudevents/http/vertx/impl/BinaryVertxMessageImpl.java:54]: Shouldn't be toLowerCase used here?

**Comment by slinkydeveloper** [comment]
@matejvasek fixed, now it works :smile:

**Pull request merged**